kolibri/core/assets/src/views/buttons-and-links/k-button.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+
+  <button
+    ref="button"
+    :class="buttonClasses"
+    :type="type"
+    :disabled="disabled"
+    @click="handleClick">
+    {{ text }}
+  </button>
+
+</template>
+
+
+<script>
+
+  import buttonClassesMixin from './buttonClassesMixin.js';
+
+  /**
+   * The kButton component is used to trigger actions
+   */
+  export default {
+    name: 'kButton',
+    mixins: [buttonClassesMixin],
+    props: {
+      /**
+       * Button label text
+       */
+      text: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Whether or not button is disabled
+       */
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * HTML button 'type' attribute (e.g. 'submit', 'reset')
+       */
+      type: {
+        type: String,
+        default: 'button',
+      },
+      /**
+       * Button appearance: 'raised', 'flat', or 'link'
```

Comment:
We are already using the word `link` to describe the type of component - `k-external-link` `k-router-link`. Is there another word we can use instead of `link` to describe the appearance? Simple? Basic?

Reply:
how about `text-link`?

Reply:
or just `text`

Reply:
Trying to avoid the word `link`. They all have text. :/ What about naming the prop `style` and the the options would be `raised`, `flat`, and `none`?

Reply:
> What about naming the prop `style`

That was my first choice, but it conflicts with the real `style` HTML attribute :)

I agree it's confusing that the word 'link' can refer to either the component type and the visual style depending on context. But I also think that it's the most descriptive way to talk about something that looks like this:

> ![image](https://user-images.githubusercontent.com/2367265/31413722-a143b75c-adcf-11e7-837d-cafdbc29fef2.png)

Note that the appearance attribute is used for all of these components, and I think it's important to drive home that a `k-button` can have the appearance of a link, and a `k-router-link` can have the appearance of a button.

How about this - the `appearance` options get more explicit for all three types:

* `raised-button`
* `flat-button`
* `basic-link`

This clearly indicates that both 'button' and 'link' are being used here to refer to a visual appearance and _not_ a component type?

Reply:
I like that much better! :take-my-money: